### PR TITLE
DAOS-5696 Agent: Enable Agent to communicate to the Control Plane to …

### PR DESCRIPTION
**Pull request by dpquigl**
…clean up handles

When the agent detects that a process has closed without disconnecting its
handles or has died before being able to do so the agent needs to relay this
information to the control plane. The PoolEvict command has been extended to
take an optional set of handles to evict without evicting the entire pool. This
change enabled the leaked handle detection and modifies the PoolEvict call to
the control plane. However the PoolEvict call into the io server does not yet
make use of this handle list for performing selective eviction.

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

**Review by mjmac — COMMENTED** [review]
This looks fine, but if we land it without the server side changes will it result in non-selective pool evictions?

**Comment by dpquigl** [comment]
@mjmac That is correct if we land this without the selective pool handle eviction killing any process connected to a pool will evict all of the handles related to the pool. If you want I can merge both tickets into one PR as separate commits so they stay distinct in the commit history when we go to land it.

**Comment by mjmac** [comment]
> 
> 
> @mjmac That is correct if we land this without the selective pool handle eviction killing any process connected to a pool will evict all of the handles related to the pool. If you want I can merge both tickets into one PR as separate commits so they stay distinct in the commit history when we go to land it.

Hmm. Well, we can't really land things that way. All of the commits in a PR are squashed down as part of the merge. I think it would be safest if we made the pool evict call from the agent a no-op in this PR and then the next PR enable it for real. My concern is that there could in theory be a significant amount of time between PR landings. Even though I think that possibility is unlikely, I'd rather not have that kind of behavior enabled on master.

**Comment by dpquigl** [comment]
@mjmac Maybe we get both PRs in a state when then can land back to back and just leave it up to @johannlombardi To make sure they both land at the same time? Once I address Ken's comments in the second PR then it should be good to go as well. The alternative is I squash it all into one Commit and just put both ticket numbers in the title and retitle the commit something that applies to both tickets. That way it can land all together.

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 1]
```diff
@@ -223,7 +223,9 @@ def __str__(self):
                      'auth_cred_to_iov': ('packed'),
                      'd_sgl_init': ('sgl->sg_iovs'),
                      'daos_csummer_alloc_iods_csums': ('buf'),
-                     'get_pool_svc_ranks': ('req')}
+                     'get_pool_svc_ranks': ('req'),
+                     'send_monitor_Request': ('reqb'),
```
Comment (ashleypittman) [inline src/tests/ftest/cart/util/cart_logtest.py:227]:
```suggestion
                     'send_monitor_request': ('reqb'),
```

**Review by tanabarr — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Comment by mjmac** [comment]
The NLT failure appears to be due to a segfault on shutdown, which was fixed on master yesterday.

**Comment by ashleypittman** [comment]
> The NLT failure appears to be due to a segfault on shutdown, which was fixed on master yesterday.

The build has been deleted now so it's not possible to tell, but this PR both fixes two warnings and adds two new ones, so it would need force-landing however David and I had discussed these already and they are expected.

**Pull request merged**